src/components/reusable/table/table.ts

```diff
@@ -23,46 +23,58 @@ export class Table extends LitElement {
   /**
    * checkboxSelection: Boolean indicating whether rows should be
    * selectable using checkboxes.
+   * @type {boolean}
+   * @default false
    */
   @property({ type: Boolean })
-  checkboxSelection = false;
+  checkboxSelection?: boolean;
```

Comment:
Boolean props should always default to false. Is there a specific reason you might need them to remain undefined?

Reply:
The updated function is triggered initially when values transition from undefined to their default settings, even if these values aren't explicitly set by the user. To avoid this, I require the default values to remain undefined rather than being automatically set to false.

Reply:
You can avoid this instead by checking:
```js 
if (changedProps.has('prop') && changedProps.get('prop') !== undefined) {
  // logic
}
```

Reply:
Tried a few solutions including this. It didn't work. When true is passed over making this.prop is true but changedProps.get('prop') at that time has a value of undefined making the condition not matched.